runtime/src/bank.rs

```diff
@@ -2622,15 +2633,14 @@ impl Bank {
         };
 
         let capitalization = self.capitalization();
-        let validator_rewards =
-            (validator_rate * capitalization as f64 * epoch_duration_in_years) as u64;
+        let validator_rewards = self.inflation_allocated_during_epoch();
```

Comment:
The function you factored out has different behavior than what the original code was doing and now we have a lot of duplicated code within `update_rewards_with_thread_pool` and `inflation_allocated_during_epoch`. What you had before was better, but feel free to take another shot at cleaning this up

Reply:
Good catch! I took another stab at organizing the sub-function. The difficulty in getting something that works well for both functions boils down to 2 main things:

1. In the test, we need most of the inflation data from bank1, but we need the pre-rewards capitalization (i.e. bank0 capitalization). In `update_rewards_with_thread_pool`, we can conveniently find the spot before the validator rewards have been applied.
2. A bunch of intermediate variables (e.g. validator/foundation inflation rate, epoch duration) are needed for logging in `update_rewards_with_thread_pool ` whereas we can discard these in the test.

If this looks too ugly, we can revert back to just updating the logic inside the test function

Reply:
Yeah, it's definitely more difficult than it seemed, thanks for digging in!